sources/api/schnauzer/src/helpers.rs

```diff
@@ -907,6 +910,128 @@ pub fn goarch(
     Ok(())
 }
 
+handlebars_helper!(is_ipv4: |cidr: str| {
+    let ip_part = validate_and_parse_cidr(cidr)?;
+    let ip = IpAddr::from_str(ip_part);
+
+    match ip {
+        Ok(IpAddr::V4(_)) => true,
+        Ok(IpAddr::V6(_)) => false,
+        Err(_) => panic!("Invalid CIDR or IP address: {}", cidr),
```

Comment:
We prefer to avoid explicit panics (`panic`, `expect`, `unwrap`) and instead use `Result` values. You could try something like this?

```suggestion
    let ip = IpAddr::from_str(ip_part).context(error::InvalidIPAddressSnafu)?;
    
    matches!(ip, IpAddr::V4(_))
```